source/book/include/marlin/book/configs/Base.h

```diff
@@ -309,6 +309,19 @@ namespace marlin {
 
 
     } // end namespace types
+
+    namespace details {
+      template<typename I, typename O>
+      constexpr O safe_cast(const I& input) {
+        if ( 
+            static_cast<O>(input) >= std::numeric_limits<O>::min()
+            && static_cast<O>(input) <= std::numeric_limits<O>::max()) {
+          return static_cast<O>(input);
+        }
+        throw "numeber cast with data lose!";
+      }
+      
+    } // end namespace details
```

Comment:
save casting of numbers to smaller spaces, 
ROOT use int for size types which don't fit with std who used size_t (unsigned long)